recipes/freealut/all/patches/freealut-1.1.0-conan-build.patch

```diff
@@ -0,0 +1,184 @@
++SET(PACKAGE_BUILD_VERSION "0")
+ SET(PACKAGE_VERSION "${PACKAGE_MAJOR_VERSION}.${PACKAGE_MINOR_VERSION}.${PACKAGE_BUILD_VERSION}")
+ SET(MAJOR_VERSION "0")
+ SET(MINOR_VERSION "0")
+@@ -36,14 +38,14 @@ SET(ALUT_SOURCES src/alutBufferData.c
+                  src/alutVersion.c
+                  src/alutWaveform.c)
+ 
+-INCLUDE_DIRECTORIES(${Alut_SOURCE_DIR}/include)
++INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR}/include)
+ 
+ SET(ADD_WFLAGS "")
+ SET(ADD_CFLAGS "")
+ SET(ADD_LDFLAGS "")
+ SET(ADD_LIBS "")
+ 
+-OPTION(BUILD_STATIC    "build static library too"   ON)
++option(BUILD_SHARED_LIBS "Build using shared libraries" ON)
+ OPTION(PROFILE         "enable profile"            OFF)
+ OPTION(OPTIMIZATION    "enable optimization"        ON)
+ OPTION(WARNINGS        "enable warnings"            ON)
+@@ -72,24 +74,9 @@ ADD_DEFINE("__NO_CTYPE 1")
+ ADD_DEFINITIONS(-DHAVE_CONFIG_H)
+ ADD_DEFINITIONS(-DNDEBUG)
+ 
+-FIND_LIBRARY(OPENAL_LIB NAMES openal openal32 PATHS /usr/lib /usr/local/lib ${OPENAL_LIB_DIR})
+-IF(OPENAL_LIB MATCHES "NOTFOUND")
+-  MESSAGE(FATAL_ERROR "OpenAL not installed, cannot build alut - aborting.")
+-ENDIF(OPENAL_LIB MATCHES "NOTFOUND")
+-
+-IF(UNIX)
+- SET(ADD_LIBS ${ADD_LIBS} m)
+-ENDIF(UNIX)
+-
+-SET(CMAKE_REQUIRED_INCLUDES ${OPENAL_INCLUDE_DIR})
+-CHECK_INCLUDE_FILES("AL/alc.h;AL/al.h" AL_HEADERS)
+-IF(NOT AL_HEADERS)
+-  MESSAGE(FATAL_ERROR "OpenAL header files not found - aborting.")
+-ENDIF(NOT AL_HEADERS)
+ 
+-IF(DEFINED OPENAL_INCLUDE_DIR)
+-  INCLUDE_DIRECTORIES(${OPENAL_INCLUDE_DIR})
+-ENDIF(DEFINED OPENAL_INCLUDE_DIR)
++FIND_PACKAGE(OpenAL REQUIRED)
++set(OPENAL_LIB "OpenAL::OpenAL")
+ 
+ FIND_PACKAGE(SleepFunction)
+ 
+@@ -131,30 +118,19 @@ SET(CMAKE_CXX_FLAGS "${ADD_CFLAGS} ${CMAKE_CXX_FLAGS}")
+ SET(CMAKE_SHARED_LINKER_FLAGS "${ADD_LDFLAGS} ${CMAKE_SHARED_LINKER_FLAGS}")
+ SET(CMAKE_MODULE_LINKER_FLAGS "${ADD_LDFLAGS} ${CMAKE_MODULE_LINKER_FLAGS}")
+ 
+-IF(BUILD_STATIC)
+-  # we can't create a static library with the same name
+-  # as the shared one, so we copy it over after creation
+-  ADD_LIBRARY(alut_static STATIC ${ALUT_SOURCES})
+-  TARGET_LINK_LIBRARIES(alut_static ${OPENAL_LIB} ${ADD_LIBS})
+-  IF(NOT WIN32)
+-    ADD_CUSTOM_COMMAND(
+-      TARGET alut_static
+-      POST_BUILD
+-      COMMAND ${CMAKE_COMMAND}
+-      ARGS -E copy
+-        ${CMAKE_BINARY_DIR}/${CMAKE_STATIC_LIBRARY_PREFIX}alut_static${CMAKE_STATIC_LIBRARY_SUFFIX}
+-        ${CMAKE_BINARY_DIR}/${CMAKE_STATIC_LIBRARY_PREFIX}alut${CMAKE_STATIC_LIBRARY_SUFFIX})
+-    INSTALL_FILES(/lib FILES ${CMAKE_STATIC_LIBRARY_PREFIX}alut${CMAKE_STATIC_LIBRARY_SUFFIX})
+-  ENDIF(NOT WIN32)
+-ENDIF(BUILD_STATIC)
+-
+-
+-ADD_LIBRARY(alut SHARED ${ALUT_SOURCES})
++ADD_LIBRARY(alut ${ALUT_SOURCES})
+ SET_TARGET_PROPERTIES(alut PROPERTIES VERSION ${VERSION} SOVERSION ${MAJOR_VERSION})
++set_target_properties(alut PROPERTIES PUBLIC_HEADER "include/AL/alut.h")
++if (NOT BUILD_SHARED_LIBS)
++  target_compile_definitions(alut PUBLIC ALUT_LIBTYPE_STATIC)
++endif()
```

Comment:
Not needed. `add_library()` uses the value of `BUILD_SHARED_LIBS` automatically to set `STATIC`/`SHARED`.

Reply:
this define is being used in the alut.h to remove the dllexport/dllimport